controller/controller_error.py

```diff
@@ -0,0 +1,5 @@
+class ControllerError(Exception):
+
+    def __init__(self, message):
+        self.message = message
+        super().__init__(message)
```

Comment:
I think this should be in the main branch now? 
Maybe best to rebase/merge the latest main branch.
(merge for less hassles.)